controller/run.go

```diff
@@ -6,147 +6,71 @@ package controller
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"net"
 	"os"
 	"os/signal"
-	"sync"
 	"syscall"
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
-
 	"github.com/elastic/elastic-agent-client/v7/pkg/client"
 	"github.com/elastic/elastic-agent-libs/logp"
 	"github.com/elastic/elastic-agent-shipper/config"
-	"github.com/elastic/elastic-agent-shipper/monitoring"
-	"github.com/elastic/elastic-agent-shipper/output"
-	"github.com/elastic/elastic-agent-shipper/queue"
-	"github.com/elastic/elastic-agent-shipper/server"
-
-	pb "github.com/elastic/elastic-agent-shipper-client/pkg/proto"
 )
 
 // LoadAndRun loads the config object and runs the gRPC server
 func LoadAndRun() error {
-	agentClient, _, err := client.NewV2FromReader(os.Stdin, client.VersionInfo{Name: "elastic-agent-shipper", Version: "v2"})
-	if err != nil {
-		return fmt.Errorf("error reading control config from agent: %w", err)
+	cfg, err := config.ReadConfigFromFile()
+	switch {
```

Comment:
But in case you need to test several configurations you also need a configuration filename. I figured it would be easier to just base the managed mode on a single flag.